### PR TITLE
Development Version 1.0

### PR DESCRIPTION
**Pull request by supminn**
Most of the functionalities mentioned on [ReadME](https://github.com/supminn/neoG_eCommerce/blob/development/ReadME.md) have been implemented.

### CODE REVIEW

**Comment by MdBilal420** [comment]
Here item in cart is 1 and quantity is 3
In number of cart items it is showing quantity of items. Do Check

File: src/Components/Cart/addToCart.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { useDataContext } from "../../Context/data-context";
+
+const itemExistsInCart = (cartItems, productId) => 
+    cartItems.some(cartItem => cartItem.id === productId && cartItem.quantity>0);
+
+
+
+export const AddToCart = ({product}) => {
```
Comment (agauravdev) [inline src/Components/Cart/addToCart.js:8]: I don't think you need a react fragement here.

File: src/Reducer/data-reducer.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
+export const dataReducer = (state, { type, payload }) => {
```
Comment (agauravdev) [inline src/Reducer/data-reducer.js:1]: Could use enums.

**Review by agauravdev — COMMENTED** [review]
Great code supriya. I don't have much to say. Just a couple minor things which don't matter much as of now.

File: src/Reducer/data-reducer.js (hunk 2 above)
Comment (supminn) [reply src/Reducer/data-reducer.js:1]: I need to learn more about enums, will do it once I am comfortable.

File: src/Components/Cart/cart.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+              onClick={() =>
+                dispatch({
+                  type: "SHOW_TOAST",
+                  payload: "functionality coming soon...",
+                })
+              }
+            >
+              Place Order
+            </button>
+          </section>
```
Comment (payalkherajani) [inline src/Components/Cart/cart.js]: You can make a separate component for price details

File: src/Components/Toast/toast.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { useEffect } from "react";
+import { useDataContext } from "../../Context/data-context";
+
+export const Toast = () => {
+  const {state:{toastMsg}, dispatch} = useDataContext();
+
+  const  closeToast = () => {
+    console.log(toastMsg); //undefined -- FIX IT
```
Comment (payalkherajani) [inline src/Components/Toast/toast.js]: you can remove console here

File: src/Components/Toast/toast.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  useEffect(() => {
+    const timeID = setTimeout(closeToast ,2000);
+    return () => clearTimeout(timeID);  
+  })
+
+  return (
+    <div className="toast-container">
+      <h3 className="toast-message">{toastMsg}</h3>
+      <i onClick={closeToast} className="fa fa-times wish-remove"></i>
+    </div>
```
Comment (payalkherajani) [inline src/Components/Toast/toast.js:20]: Show your toast at top right corner instead of bottom right

**Comment by payalkherajani** [comment]
Very well written code!

File: src/Components/Toast/toast.js (hunk 4 above)
Comment (supminn) [reply src/Components/Toast/toast.js]: Thanks for this observation. I had missed it completely

**Pull request merged**